include/core/shape.h

```diff
@@ -757,6 +839,190 @@ class CompositeShape : public Shape<dim>
     operations;
 };
 
+template <int dim>
+class OpenCascadeShape : public Shape<dim>
```

Comment:
Can you write unit tests for this shape ? It would be important to have unit tests to just test the distance calculation instead of relying on the application test so much.